modin/engines/base/frame/data.py

```diff
@@ -976,37 +976,83 @@ def _join_index_objects(self, axis, other_index, how, sort):
 
         Parameters
         ----------
-            axis : 0 or 1
-                The axis index object to join (0 - rows, 1 - columns).
-            other_index : Index
-                The other_index to join on.
-            how : {'left', 'right', 'inner', 'outer'}
-                The type of join to join to make.
-            sort : boolean
-                Whether or not to sort the joined index
+        axis : 0 or 1
+            The axis index object to join (0 - rows, 1 - columns).
+        others_index : list(Index)
+            The others_index to join on. Index of `self` frame should be first.
```

Comment:
why should we pass `self` indices in a list with `others_index`? When we do `self._join_index_objects` we can retrieve `self` indices from `self.index`. Or we can mark this method as 'static' because we're not using `self` inside it at all

Reply:
There is no specificity in the context of `self._join_index_objects` for `self.index`.  Besides, in the case of duplicates, we get 2 lists: a list of indexes and a corresponding list of indexers (`others_index[i] -> indexers[i]`).

Make sense to have staticmethod decorator for the function.